eclipsesource/theia
# [vscode] support CommentThread state property

#### What it does

_Add Comment#state optional property_
This adds the state optional property from Comment to improve vscode api coverage. It also displays the state in the Comment Thread widget

Fixes #12231 

Contributed on behalf of ST Microelectronics

#### How to test 
Note: there are currently some issues with menus, since theia 1.28.0. This is due to #11290, integrated in 1.28.0. Prior to this change, the comment sample works on theia 1.27.0. So to test this change, I cherry-picked my commit on top of 1.27.0 tag. 
This is the same case as for PR https://github.com/eclipse-theia/theia/pull/12007.
1. Install forked extension [#comment-sample-0.0.1](https://github.com/rschnekenbu/vscode-extension-samples/releases/download/comment-with-state/forked-comment-api-example-0.0.1.vsix) vsix file - [Source](https://github.com/rschnekenbu/vscode-extension-samples/tree/rschnekenbu/comment-state/comment-sample)
2. Create a comment using the side bar while comparing 2 test files and reply with some comment with toggle or reply. 
![12231](https://user-images.githubusercontent.com/3964263/233108046-dcc3e6dc-12af-4a3a-bd63-a867882ded8d.gif)
3. The resolved state should be displayed along the thread label.

#### Review checklist


#### Reminder for reviewers
